SE_Topics.md

```diff
@@ -9,13 +9,17 @@ Please feel free to submit a PR with suggestions of topics to add.
 * Version control
 * Automated testing
 * Code review
-* Specification Writing
-* Code smells, naming practices, code style etc ?
+* Requirements Engineering
```

Comment:
What do you mean by "requirements engineering"?  (Example, link, etc?)

Reply:
By Requirements Engineering I mean the activity of eliciting/analyzing/writing/validating/evolving requirements.

Reply:
In Google terms we would say "How to write a design doc / design proposal," I think.

Unless you mean that in the sense of how requirements are produced in other engineering disciplines: "Company A will produce a Widget with tolerances of X, Y, Z of not more than 7x4x2 Smoots, that will interface with ..."

I think the major difference between those approaches is the assumption that everything can be designed on paper and go roughly according to that design vs. "here's a problem, here are the constraints, here are possible solutions and their pros/cons, here is what the direction I think we should go."  The "Design Doc" style produces more of a consensus roadmap rather than a complete design.

(I'd be happy for other industry folks to chime in and say that my experience here is not representative.)

Reply:
@tituswinters, I like this idea of the design proposal. Please if there's a sample of such design doc, I'd be happy to learn a little more details.

Reply:
I think that would be helpful (both an example and a template).